Redesign Homepage Layout Using CSS Grid

Aeeshadam opened the pull request:
This PR introduces a redesign of the previously implemented homepage layout, replacing Flexbox with CSS Grid for layout management. The page has been restructured to meet the provided acceptance criteria and enhance layout flexibility with Grid-based positioning. 

CSS Grid is used to replace Flexbox in all major sections of the page, providing a more structured and flexible grid-based layout.

Aeeshadam committed:
created a page layout with flexbox

Aeeshadam committed:
redesigned previous page using grid layout

Aeeshadam committed:
removed unused class

Aeeshadam committed:
added max width and rename spacing variables

aayusachdev commented:
Hey @Aeeshadam since the React JS code for this PR is similar to the flexbox one, it would be nice if you can address those comments in these files as well, then we can review the CSS Grid changes separately.

Aeeshadam committed:
refactor(home): restructure folder layout and adjust naming convention

Aeeshadam committed:
refactor(footer, navbar, services, constants, hero): implement review suggestions
- Footer: destructured  and  objects; added optional chaining for safe access
- Navbar: added optional chaining to handle undefined values safely
- Services: added optional chaining and conditional rendering to ensure stability
- Constants: added unique id for all data to be used as key in lists
- Hero: removed button wrapping anchor

Aeeshadam commented:
Thanks for the feedback @aayusachdev

I have made all the suggested changes in this branch and the responsive design branch, kindly review at your earliest convenience.

Thanks

aayusachdev commented on src/index.css:
```diff
@@ -1,13 +1,46 @@
-body {
+:root {
+  --grey-color: rgb(58, 58, 58);
```
where are you importing this index.css file?

Aeeshadam replied:
Importing it in index.js file in the src directory

aayusachdev commented:
Review completed.
